### PR TITLE
mrc-1685 Added buildkite pipeline

### PR DESCRIPTION
**Pull request by EmmaLRussell**
I think I mopped up most of the references to the private docker registry in the course of doing this, though there may still be some in the tc scripts. I've given buildkite its own version of the scripts that tc uses, and marked those as deletable once tc is retired.

### CODE REVIEW

File: buildkite/common [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+# Export env vars needed for running test dependencies
+export REGISTRY=vimc
+export MONTAGU_ORDERLY_SERVER_VERSION=$(<$HERE/../config/orderly_server_version)
+export GIT_ID=$GIT_ID
+export GIT_BRANCH=$GIT_BRANCH
+export MONTAGU_ORDERLY_PATH=$PWD/git
+export ORDERLY_SERVER_USER_ID=$UID
+
+function cleardocker() {
+  $here/../scripts/clear-docker.sh
```
Comment (EmmaLRussell) [inline buildkite/common:34]: This is a bit of a sledgehammer, but should do the job.

File: buildkite/pipeline.yml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+      command: buildkite/make-migrate-image.sh
+
+    - wait
+
+    - label: ":mag: Build and test app"
+      command: buildkite/build-app.sh
+
+    - wait
+
+    - label: ":mag: Run smoke test"
```
Comment (EmmaLRussell) [inline buildkite/pipeline.yml:12]: Not sure if this is the right level of parallel steps? e.g. could do smoke test before custom config tests.

File: buildkite/common [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+#   dependabot/npm_and_yarn/app/lodash-4.17.19
+#
+# But docker does not like
+GIT_BRANCH=$(echo $GIT_BRANCH | sed 's;/;-;g')
+
+BUILDKITE_DOCKER_AUTH_PATH=/var/lib/buildkite-agent/.docker/config.json
+
+# Export env vars needed for running test dependencies
+export REGISTRY=vimc
+export MONTAGU_ORDERLY_SERVER_VERSION=$(<$HERE/../config/orderly_server_version)
```
Comment (hillalex) [inline buildkite/common:27]: `$HERE` isn't defined, so this isn't pulling in the version as desired. From buildkite logs: 
`./buildkite/common: line 27: /../config/orderly_server_version: No such file or directory`

**Review by hillalex — COMMENTED** [review]
Nice one :tada:  Could you also update the README to replace the Teamcity section with a Buildkite section?

Couple of things to do next I think (once this is merged):
- go through and remove the `TeamcityTest` class
- fix up those template tests so they don't barf a million lines of output - noticed the Buildkite logs are unviewable because of all the excess output!

**Comment by EmmaLRussell** [comment]
Added tickets for these, plus removing the TeamCity build:
https://mrc-ide.myjetbrains.com/youtrack/issue/mrc-1828
https://mrc-ide.myjetbrains.com/youtrack/issue/mrc-1829
https://mrc-ide.myjetbrains.com/youtrack/issue/mrc-1830

> Couple of things to do next I think (once this is merged):
> 
>     * go through and remove the `TeamcityTest` class
> 
>     * fix up those template tests so they don't barf a million lines of output - noticed the Buildkite logs are unviewable because of all the excess output!

File: buildkite/run-custom-config-tests-in-container.sh [hunk 4]
```diff
@@ -0,0 +1,31 @@
+#!/usr/bin/env bash
+set -ex
+here=$(dirname $0)
+. $here/common
+
+# Make the build environment image that is shared between multiple build targets
+$here/make-build-env.sh
```
[the remark this answers is not in the document]
Comment (EmmaLRussell) [reply buildkite/run-custom-config-tests-in-container.sh:7]: Added a ticket for that. 
https://mrc-ide.myjetbrains.com/youtrack/issue/mrc-1832
The front end tests are pretty quick compared to the back end ones, but every little helps..

File: buildkite/run-custom-config-tests-in-container.sh [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+    -f customConfigTests.Dockerfile \
+	.
+
+# create the db
+$here/make-db.sh
+
+## Run all dependencies
+$here/../scripts/run-dependencies.sh
+
+export NETWORK=montagu_default
```
[the remark this answers is not in the document]
Comment (EmmaLRussell) [reply buildkite/run-custom-config-tests-in-container.sh]: Yes, and then it's needed by the cli. Will try adding the test user in run-deps, might break a test.

[the remark this answers is not in the document]
Comment (EmmaLRussell) [reply buildkite/run-custom-config-tests-in-container.sh]: That seemed to work :+1:

**Review by hillalex — APPROVED** [review]
Awesome, thanks :+1:

**Pull request merged**